producer/src/contract/contract_test.go

```diff
@@ -136,184 +137,230 @@ func TestYieldSerialization(t *testing.T) {
 		t.Errorf("Yield indeces do not match")
 	}
-	if !cmp.Equal(testClaim.PublicKey, testPubkey) {
+	if !reflect.DeepEqual(testClaim.PublicKey, pubKey) {
 		t.Errorf("Public Keys do not match")
 	}
-	// Case 3
-	_, shouldNotBeNil := MakeClaim("testDB.dat", testPubKey, 1)
-	if shouldNotBeNil == nil {
-		t.Errorf("Made claim on empty yield pool")
-	}
+
 }
 
-func TestMakeClaimCase2(t *testing.T) {
-	setUpDB("testDB.db")
-	defer tearDown("testDB.db")
-	testPubKey := generatePubKey()
-	testYield := MakeYield(testPubKey, 10000000)
-	contractHash := block.HashSHA256([]byte{'b', 'l', 'k', 'c', 'h', 'a', 'i', 'n'})
-	err := InsertYield(testYield, "testDB.dat", 35, contractHash, 1)
+func TestMakeClaimDeficit(t *testing.T) {
+	db := "testDB.db"
+	setUpDB(db)
+	defer tearDown(db)
+
+	amt := uint64(500)
+	pubKey := generatePubKey()
+	yield := MakeYield(&pubKey, amt)
+	contractHash := block.HashSHA256([]byte("blokchain"))
+	blockHeight := uint64(35)
+	yieldIndex := uint16(1)
+
+	err := InsertYield(yield, db, blockHeight, contractHash, yieldIndex)
 	if err != nil {
-		t.Errorf("Failed to insert yield")
+		t.Errorf("Failed to insert yield: %v", err)
 	}
-	testClaim, err := MakeClaim("testDB.dat", testPubKey, 50000)
+
+	testClaim, err := MakeClaim(db, pubKey, amt+5)
 	if err == nil {
-		t.Errorf("Should have an error that resembles change")
+		t.Errorf("No change error struct returned")
+	} else {
+		if deficit, ok := err.(DeficitError); ok {
+			if deficit.Deficit != uint64(5) {
+				t.Errorf("Wrong deficit value, expected: 5, actual: %d", deficit)
+			}
+		} else {
+			t.Errorf("Different error given: %v", err)
+		}
 	}
+
 	if !bytes.Equal(testClaim.PreviousContractHash, contractHash) {
 		t.Errorf("Contract hashes do not match")
 	}
-	if testClaim.BlockIndex != 35 {
+	if testClaim.BlockIndex != blockHeight {
 		t.Errorf("Block indeces do not match")
 	}
-	if testClaim.YieldIndex != 1 {
+	if testClaim.YieldIndex != yieldIndex {
 		t.Errorf("Yield indeces do not match")
 	}
-	if !cmp.Equal(testClaim.PublicKey, testPubkey) {
+	if !reflect.DeepEqual(testClaim.PublicKey, pubKey) {
 		t.Errorf("Public Keys do not match")
 	}
-	// Case 3
-	_, shouldNotBeNil := MakeClaim("testDB.dat", testPubKey, 1)
-	if shouldNotBeNil == nil {
-		t.Errorf("Made claim on empty yield pool")
-	}
+
 }
 
-func TestMakeClaimPriorityCase1(t *testing.T) {
-	setUpDB("testDB.db")
-	defer tearDown("testDB.db")
-	testPubKey := generatePubKey()
-	testYieldFloor := MakeYield(testPubKey, 10000000)
-	testYieldCeiling := MakeYield(testPubKey, 10000010)
-	contractHashFloor := block.HashSHA256([]byte("This should not be claimed"))
-	contractHashCeiling := block.HashSHA256([]byte("This should be claimed"))
-	err := InsertYield(testYieldFloor, "testDB.dat", 35, contractHashFloor, 1)
+func TestMakeClaimMultipleYield(t *testing.T) {
+	db := "testDB.db"
+	setUpDB(db)
+	defer tearDown(db)
+
+	amt := []uint64{uint64(100), uint64(200)}
+	pubKey := []ecdsa.PublicKey{generatePubKey(), generatePubKey()}
```

Comment:
Why are we generating two separate public keys if all these yields should be claimable by the same person?

Reply:
It's a test, MakeClaim should claim the yields corresponding to the public key it was given as a parameter.

Reply:
I don't think there would be an instance where a client has yields corresponding to someone else...

Also what happened to prioritizing yields that are min(abs(yieldval - desiredval), ... ) ?